modules/apps/frontend-js/frontend-js-a11y-web/src/main/resources/META-INF/resources/Violation.tsx

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import './Violation.scss';
+
+import ClayList from '@clayui/list';
+import ClayPanel from '@clayui/panel';
+import React from 'react';
+
+import PanelNavigator from './PanelNavigator';
+import Rule from './Rule';
+
+import type {Result} from 'axe-core';
+
+declare var Liferay: {
+	Language: {
+		get(value: string): string;
+	};
+};
+
+type PanelSectionProps = {
+	children: React.ReactNode;
+	title: React.ReactNode;
+};
+
+function PanelSection({children, title}: PanelSectionProps) {
+	return (
+		<ClayPanel
+			displayTitle={title}
+			displayType="unstyled"
+			role="tab"
+			showCollapseIcon={false}
+		>
+			<ClayPanel.Body>{children}</ClayPanel.Body>
+		</ClayPanel>
+	);
+}
+
+type TViolationNext = {
+	occurrenceIndex: number;
+	occurrenceName: string;
+	violationIndex: number;
+};
+
+type ViolationProps = {
+	navigationState?: {violationIndex: number};
+	next?: (payload: TViolationNext) => void;
+	previous?: () => void;
```

Comment:
Are you putting this `optional` because we declared it in `A11yPanel` right? maybe we should find a way in the future for this to be required...

Reply:
Yes :/ for typing purposes since it's currently rendered by `React.Children.map`

Reply:
Yeah, well I think we are going to have to live with this, I have no idea how we could avoid this in dynamic rendering. Perhaps using generic types or `Required` types...